contracts/swap/swap.go

```diff
@@ -47,6 +47,10 @@ type Backend interface {
 
 // Contract interface defines the methods exported from the underlying go-bindings for the smart contract
 type Contract interface {
+	// Withdraw attempts to withdraw Wei from the chequebook
+	Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error)
```

Comment:
for consistency's sake, you should do away with the `amount` parameter here, just as you did in the `Deposit` function

Reply:
That is not possible here. In Ethereum, a transaction is comprised of `value` and `callData` (among others). When sending a TX with value (such as deposit). Geth expects the `callData` as arguments, while all other info needed is under the `TransactOpts`. The `amount` in this function is part of the `callData` and hence cannot be removed from the `Withdraw` function call.

Reply:
i see. that being the case, i'd rather you added `amount` back to `Deposit`.

we can either:
- ignore whatever value is set in the `opts`/`auth` variable (or even ignore whether we have one or not).
- output an error if 2 values are submitted, so long as they don't match. 

in any case, i think it's worth having these 2 methods structurally equivalent.

more opinions welcome, as usual

Reply:
Added `amount` back to `Deposit` and did proposed check on `amount` and auth.Value.